src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -582,6 +596,10 @@ abstract class ManagedStrategy(
         if (suspendedThread != null) {
            return suspendedThread
         }
+        // if the thread switch is optional, continue with the current thread
+        if (!mustSwitch) {
```

Comment:
Why is this check necessary?

Reply:
It was in the code before, I only moved it below, after handling other cases first.

In general, this case occurs when strategy attempts a to make a thread switch, all other threads are blocked or finished, but the thread we are switching from is not blocked. Thus it is fine to continue executing the current thread.

Reply:
Reverted